src/ts/components/extensions/richtexteditor/RichTextEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import { BoxProps } from "props/box";
+import { DashBaseProps } from "props/dash";
+import { StylesApiProps } from "props/styles";
+import React, { Suspense } from "react";
+
+// eslint-disable-next-line no-inline-comments
+const LazyRichTextEditor = React.lazy(() => import(/* webpackChunkName: "RichTextEditor" */ './fragments/RichTextEditor'));
+
+/** Supported extensions (add more as needed). */
+type Extension = "StarterKit" | "Underline" | "Underline" | "Link" | "Superscript" | "SubScript" | "Highlight" | "TextAlign" | "Table" | "TableCell" | "TableRow" | "TableHeader";
+
+/** Supported controls (add more as needed). */
+type Control = "Bold" | "Italic" | "Underline" | "Strikethrough" | "ClearFormatting" | "Highlight" | "Code" | "H1" | "H2" | "H3" | "H4" | "Blockquote" | "Hr" | "BulletList" | "OrderedList" | "Subscript" | "Superscript" | "Link" | "Unlink" | "AlignLeft" | "AlignCenter" | "AlignJustify" | "AlignRight" | "Undo" | "Redo";
+
+/** Toolbar property definition. */
+type Toolbar = {
+  sticky?: boolean;
+  controlsGroups?: Control[][];
+}
```

Comment:
Would it be better to define these types in the props folder?

Reply:
As a general rule, I would typically hold models definitions (here, props) close to their usage (i.e. here) when they are only used in a single place, and only move them when they start to be used in multiple places (like e.g. `DashBaseProps`). That being said, I am happy to follow any style guides for the library. Would you like them moved?